### PR TITLE
Prioritize anchor

### PR DESCRIPTION
**Pull request by mikemaccana**
- Add a general 'intro to onchain' lesson with an anchor/native agnostic summary.
- Make 'local setup' lesson set up a machine rather than deploy a native program. Just point to Microsoft's dos for WSL, they'll be more up to date.
- Put these two lessons into a 2 unit 'Introduction to Onchain Program Development' track, that then tells people to pick between Anchor and Native
- Reorder native and anchor tracks
- Rename 'hello world' chapter to 'hello world in native'

### CODE REVIEW

File: content/intro-to-onchain.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+- **On chain programs** are programs that run on Solana. 
+- Programs have one or more **instruction handlers**. Instruction handlers are functions that process the **instructions** in Solana transactions.
+- Instruction handlers write their data to Solana **accounts** and then either succeed or fail. 
+- Solana programs are most commonly written in **Rust**, typically with the **Anchor** framework.
+- Anchor programs have IDLs, small JSON files which can be used to automatically create JS/TS code to build instructions for clients.  
+
+# Overview
+
+Each Solana cluster (mainnet-beta, testnet, devnet, localnet) is effectively a single computer with globally syncronized state. The programs that run on Solana - the ones that create tokens, swap tokens, art marketplaces, escrows, market makers, DePIN apps, auctions, retail payments platforms, etc - are called **Solana apps**.
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:19]: Typo `syncronized` -> `synchronized`

File: content/intro-to-onchain.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+The most popular way to build onchain apps is using **Rust** language and the **Anchor** framework. Frameworks for other languages, including **TypeScript**, **Python** and **Move** are coming soon. 
+
+These frameworks implement common security checks automatically, and handle common tasks like:
+ - sending incoming instructions to the right instruction handlers
+ - deserialising data from incoming transactions
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:25]: Would prefer the American spelling for deserialize

Comment (jamesrp13) [reply content/intro-to-onchain.md:25]: Might consider running this through something like Grammarly with those sorts of things in mind

File: content/intro-to-onchain.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ - checking the accounts provided with incoming instructions, for example to check that certain accounts are of a particular type, or are distinct from other accounts.
+
+Regardless of the language and framework you choose, Solana works the same. Let's refresh how programs work on Solana.
+
+![Diagram showing a transaction with twp instructions](../assets/transaction-and-instructions.svg)
+
+## Programs are deployed at addresses
+
+Like like we can send tokens to users using their public key, we can find programs using the program's public key. When using Anchor, a keypair is created during `anchor init` and the private key saved in the `target/deploy` directory of your project.
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:34]: Typo `Like like` -> `Like`

File: content/intro-to-onchain.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+A program's public key is sometimes called a 'program ID'.
+
+## Programs have instruction handlers
+
+For example, a Solana client making a transaction transferring some USDC with a memo saying 'thanks' would have two instructions:
+  - one instruction for the Token program's
+`transfer` instruction handler
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:42]: Is this carriage return intentional?

File: content/intro-to-onchain.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+But unlike HTTP route handlers, Solana instruction handlers don't return data. Instead, the instruction handlers write their data to accounts on Solana.
+
+Programs on Solana can transfer tokens, which end up in user wallet addresses (for SOL) for the users token accounts (for other tokens). 
+
+But more importantly, programs on Solana can create additional addresses as needed, to store items of data. 
+
+## Programs store data in Program Derived Addresses (PDAs), a key-value store
+
+These addresses are called **program derived addresses** and act as a **key / value store**.
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:61]: For the sake of scanability, I would prefer that we start each section with something that doesn't depend on the last sentence of the previous sentence, so instead of "these addresses are called...." I would reword to "Addresses created by Solana programs are called program derived addresses..."

File: content/intro-to-onchain.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+ - The 'key' is the address of the PDA, which is determined by `seeds` chosen by you, the programmer. 
+   - Want an account to store USD to AUD exchange rate? Your program can use the seeds `USD` and `AUD` to make a Program Derived Address
+   - Want to store information about the relationship of two users? You can use **both those users wallet addreesses** as seeds to make a PDA to store that information.
+   - Want an account to store Steve's review of Titanic? Your program can use Steve's **wallet address** and the string `titanic` (or maybe the IMDB ID if you prefer) to make a Program Derived Address.
+   - Want some global infomation for your entire program? You can use a string like `config`. Your program's PDAs are unique, so these won't conflict with other users. 
+ - The value is the data inside.
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:68]: Maybe for clarify we say "The value is the data inside the account at the given address" ?

Comment (jamesrp13) [reply content/intro-to-onchain.md:68]: Idk this is a weird one actually where we definitely say "it's in the PDA" even though technically PDA stands for an address not an account...idk I'm not opinionated here actually

File: content/intro-to-onchain.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+   - The data inside the PDA is determined by you, the programmer.
+
+Key value stores allow your on chain program, and client software, to consistently determine the address data can be found at, because the same seeds will always return the same address.
+
+## Solana front end apps need to specify all the accounts they will use
+
+As you may already know, Solana is fast because it can process transactions that don't overlap at the same time. Ie, just like the real world, Alice sending to Bob doesn't stop Chris from sending something to Diana. Your front end apps needs to specify the addresses of all the accounts they will use.
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:75]: `Ie` should probably be `I.e.`

**Review by jamesrp13 — CHANGES_REQUESTED** [review]
I love this as an intro. Just a couple nit picks in my comments but otherwise let's ship it!

Just a note, I didn't review the local-setup changes as that seems more procedural. If you feel strongly about me taking a look lmk and I can get to it later today or tomorrow

File: content/intro-to-onchain.md [hunk 8]
```diff
@@ -58,21 +57,21 @@ But more importantly, programs on Solana can create additional addresses as need
 
 ## Programs store data in Program Derived Addresses (PDAs), a key-value store
 
-These addresses are called **program derived addresses** and act as a **key / value store**.
+Data for Solana programmed are stored in **program derived addresses (PDAs)**. Solana's PDAs can be thought of as a **key / value store**:
```
Comment (jamesrp13) [inline content/intro-to-onchain.md:60]: typo `programmed` -> `programs`

File: content/intro-to-onchain.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+But unlike HTTP route handlers, Solana instruction handlers don't return data. Instead, the instruction handlers write their data to accounts on Solana.
+
+Programs on Solana can transfer tokens, which end up in user wallet addresses (for SOL) for the users token accounts (for other tokens). 
+
+But more importantly, programs on Solana can create additional addresses as needed, to store items of data. 
+
+## Programs store data in Program Derived Addresses (PDAs), a key-value store
+
+Data for Solana programmed are stored in **program derived addresses (PDAs)**. Solana's PDAs can be thought of as a **key / value store**:
```
Comment (mikemaccana) [inline content/intro-to-onchain.md:60]:
```suggestion
Data for Solana programs are stored in **program derived addresses (PDAs)**. Solana's PDAs can be thought of as a **key / value store**:
```
